File: 品味Java的21種設計模式/dp_java/memento/example2/Caretaker.java
package cn.javass.dp.memento.example2;
import java.util.*;
/**
 * 负责保存备忘录的对象
 */
public class Caretaker {
    /**
     * 记录被保存的备忘录对象
     */
    private Memento memento = null;
    /**
     * 保存备忘录对象
     * @param memento 被保存的备忘录对象
     */
    public void saveMemento(Memento memento) {
        this.memento = memento;
    }
    /**
     * 获取被保存的备忘录对象
     * @return 被保存的备忘录对象
     */
    public Memento retriveMemento() {
        return this.memento;
    }
}

File: 品味Java的21種設計模式/dp_cpp/memento/example2/Caretaker.cpp
#include "Caretaker.h"

namespace cn {
namespace javass {
namespace dp {
namespace memento {
namespace example2 {

void Caretaker::saveMemento(Memento *memento) {
    this->memento = memento;
}

Memento *Caretaker::retriveMemento() {
    return this->memento;
}
}
}
}
}
}
